legion/legion/k8s/enclave.py

```diff
@@ -388,6 +388,16 @@ def deploy_model(self, image, model_iam_role=None, count=1, livenesstimeout=2, r
             body=deployment,
             namespace=self.namespace)
 
+        deployment_ready = legion.utils.ensure_function_succeed(
+            lambda: image_meta_information.k8s_name in [
+                item.metadata.name
+                for item in extensions_v1beta1.list_namespaced_deployment(self.namespace).items],
+            10, 1, boolean_check=True
+        )
+
+        if not deployment_ready:
+            raise Exception('Cannot get deployment info')
```

Comment:
@kirillmakhonin use custom exception type to make custom handling logic possible in future